diff --git contiki/cpu/nrf52832/Makefile.nrf52832 contiki-nrf52-patched/cpu/nrf52832/Makefile.nrf52832
--- contiki/cpu/nrf52832/Makefile.nrf52832
+++ contiki/cpu/nrf52832/Makefile.nrf52832
@@ -163,7 +163,7 @@ CFLAGS += -DCONFIG_GPIO_AS_PINRESET
 CFLAGS += -DBLE_STACK_SUPPORT_REQD
 CFLAGS += -mcpu=cortex-m4
 CFLAGS += -mthumb -mabi=aapcs --std=gnu99
-CFLAGS += -Wall -Werror
+CFLAGS += -Wall
 CFLAGS += -ggdb
 CFLAGS += -mfloat-abi=hard -mfpu=fpv4-sp-d16
 # keep every function in separate section. This will allow linker to dump unused functions
diff --git contiki/cpu/nrf52832/ld/nrf52-pca10040-sd.ld contiki-nrf52-patched/cpu/nrf52832/ld/nrf52-pca10040-sd.ld
--- contiki/cpu/nrf52832/ld/nrf52-pca10040-sd.ld
+++ contiki-nrf52-patched/cpu/nrf52832/ld/nrf52-pca10040-sd.ld
@@ -5,8 +5,8 @@ GROUP(-lgcc -lc -lnosys)
 
 MEMORY
 {
-  FLASH (rx) : ORIGIN = 0x1f000, LENGTH = 0x61000
+  FLASH (rx) : ORIGIN = 0x2f100, LENGTH = 0x3FE00
   RAM (rwx) :  ORIGIN = 0x20002800, LENGTH = 0xD800
 }
 
-INCLUDE "nrf5x_common.ld"
\ No newline at end of file
+INCLUDE "nrf5x_common.ld"
diff --git contiki/platform/nrf52dk/contiki-main.c contiki-nrf52-patched/platform/nrf52dk/contiki-main.c
--- contiki/platform/nrf52dk/contiki-main.c
+++ contiki-nrf52dk-patched/platform/nrf52dk/contiki-main.c
@@ -110,6 +110,7 @@ static void
 board_init(void)
 {
 #ifdef SOFTDEVICE_PRESENT
+    sd_softdevice_vector_table_base_set(0x2f100);
   /* Initialize the SoftDevice handler module */
   SOFTDEVICE_HANDLER_INIT(NRF_CLOCK_LFCLKSRC_XTAL_20_PPM, NULL);
 #endif
